src/Table/index.js

```diff
@@ -84,68 +84,57 @@ StyledTable.defaultProps = {
   theme: defaultTheme,
 };
 
-class Table extends React.PureComponent<Props, State> {
-  state = {
-    showShadows: false,
-    showRight: false,
-    showLeft: false,
+const Table = ({ children, compact = false, dataTest }: Props) => {
+  const [shadows, setShadows] = React.useState(false);
+  const [right, setRight] = React.useState(false);
+  const [left, setLeft] = React.useState(false);
+
+  const outer: { current: any | HTMLElement } = React.useRef(null);
+  const inner: { current: any | HTMLElement } = React.useRef(null);
+  const table: { current: any | HTMLElement } = React.useRef(null);
+
+  const handleScroll = () => {
+    if (shadows && inner && table && outer) {
+      setLeft(inner.current?.scrollLeft >= 5);
+      setRight(
+        inner.current?.scrollLeft + outer.current?.clientWidth <= table.current?.clientWidth,
+      );
+    }
   };
 
-  outer: { current: any | HTMLElement } = React.createRef();
-
-  inner: { current: any | HTMLElement } = React.createRef();
-
-  table: { current: any | HTMLElement } = React.createRef();
-
-  componentDidMount() {
-    window.addEventListener("resize", this.handleResize);
-    this.handleResize();
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener("resize", this.handleResize);
-  }
-
-  handleResize = () => {
-    const { table, outer } = this;
+  const handleResize = () => {
     if (table && outer) {
```

Comment:
Wrap the `handleResize` into `useCallback` and use it as dependency of `useEffect`. Or declare it inside the `useEffect` and use empty dependency array `[]`.